packages/protocol/contracts/governance/Governance.sol

```diff
@@ -618,6 +620,29 @@ contract Governance is
     return true;
   }
 
+  /**
+   * @notice Unapproves a proposal only if it is in approved stage.
+   * @param proposalId The ID of the proposal to unapprove.
+   * @param index The index of the proposal ID in `dequeued`.
+   * @return Whether or not the unapproval was made successfully.
+   */
+  function unapprove(uint256 proposalId, uint256 index) external onlyApprover returns (bool) {
+    dequeueProposalsIfReady();
+    (Proposals.Proposal storage proposal, Proposals.Stage stage) = requireDequeuedAndDeleteExpired(
+      proposalId,
+      index
+    );
+    if (!proposal.exists()) {
+      return false;
+    }
+
+    require(proposal.isApproved(), "Proposal is not approved");
+    require(stage < Proposals.Stage.Execution, "Proposal not in Referendum stage");
```

Comment:
I think if we are allowing unapproval in `Referendum` we might also want to allow it in `Execution`? it can obviously be front-run though
cc @nambrot @aslawson

Reply:
@yorhodes I have made the changes to unapprove proposals that are in execution stage.